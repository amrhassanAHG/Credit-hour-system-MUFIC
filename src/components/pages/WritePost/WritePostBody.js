import React, { Component } from "react";
import { NavLink } from "react-router-dom";
import validator from 'validator';
export default class WritePost extends Component {
  constructor(props) {
    super(props);
    this.state = {
       title: '',
       imageLink: '',
       postBody: ''
    };
  }

  componentDidMount = () => {

  };

  submit = (e) => {
    e.preventDefault();
    let isValid = true ;
    const inputs = document.getElementsByClassName("writePost-item");
    for(let i = 0 ;  i < inputs.length; i++){
       if(validator.isEmpty(inputs[i].value)){
          isValid = false ;
          inputs[i].classList.add("is-invalid");
       } else inputs[i].classList.remove("is-invalid");
    }


    /*for testing*/
    // const postWrite = document.getElementsByClassName("note-editing-area")[0];
    // if(validator.isEmpty(postWrite.innerText)){
    //     isValid = false ;
    //     postWrite.classList.add("is-invalid");
    // } else postWrite.classList.remove("is-invalid");

    const postWrite = document.getElementsByClassName("note-editing-area")[0]
    this.setState((prev) => {
      return {
        ...prev,
        postBody: postWrite.innerHTML
      }
    });

    if(!isValid) return;

    //i used timeout cause the setState upove of adding postBody takes time 
    setTimeout(() => {
        //rest of work send requests
        console.log(this.state);
    }, 500)
  }

  uploadImage = (e) => {
    e.persist();
    if(!validator.isEmpty(e.target.value)) e.target.classList.remove("is-invalid");
    //  const image = document.getElementById("previewImage");
    const imageUrl = URL.createObjectURL(e.target.files[0]);
    this.setState(() => {
      return {
        imageLink: imageUrl
      }
    });
  } 

  render() {
    return (
      <div className="content-wrapper">
        {/* Content Header (Page header) */}
        <div className="content-header">
          <div className="container-fluid">
            <div className="row mb-2">
              <div className="col-sm-6">
                <h1 className="m-0 text-dark">Write post</h1>
              </div>
              {/* /.col */}
              <div className="col-sm-6">
                <ol className="breadcrumb float-sm-right">
                  <li className="breadcrumb-item">Write post</li>
                </ol>
              </div>
              {/* /.col */}
            </div>
            {/* /.row */}
          </div>
          {/* /.container-fluid */}
        </div>
        {/* /.content-header */}

        {/* Main content */}
        <div className="content">
          <div className="container-fluid">
            <div className="card card-primary">
              <div className="card-header">
                <h3 className="card-title">Write Post</h3>
              </div>
              <form className="form form-group" role="form" onSubmit={this.submit}>
                <div className="card-body">
                  <div className="mb-3">
                    <div className='row'>
                       {/* this image is for testing */}
                       {/* <div className='col'>
                          <img src='' alt='preview' id="previewImage"/>
                       </div> */}
                       <div className='col'>
                          <input type='tel' className='form-control writePost-item' placeholder='Title' 
                          onChange={(e) => {
                            e.persist(); // for unknown reson
                            if(!validator.isEmpty(e.target.value)) e.target.classList.remove("is-invalid");
                            this.setState(() => {
                              return {
                                title: e.target.value
                              }
                            });
                          }}/>
                       </div>
                       <div className='col'>
                          <div class="custom-file">
                              <input 
                                  onChange = {this.uploadImage}
                                  type="file" 
                                  accept=".jpg, .jpeg, .png"
                                  className="custom-file-input  writePost-item" 
                                  id="inputGroupFile01" 
                                  alt=''
                              />
                              <label className="custom-file-label" htmlFor="inputGroupFile01">Choose image</label>
                          </div>  
                       </div>                     
                    </div>
                    <br />
                    <textarea
                      className="textarea"
                      placeholder="Place some text here"
                      style={{
                        width: "100%",
                        height: "200px",
                        fontSize: "14px",
                        lineHeight: "18px",
                        border: "1px solid #dddddd",
                        padding: "10px",
                      }}
                    ></textarea>
                  </div>
                </div>
                <div className="card-footer">
                  <button type="submit" className="btn btn-primary">
                    Submit
                  </button>
                </div>
              </form>
            </div>
          </div>
          {/* /.row */}
        </div>
        {/* /.container-fluid */}
      </div>
    );
  }
}
