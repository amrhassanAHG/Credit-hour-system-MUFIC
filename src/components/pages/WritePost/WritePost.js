import React from 'react'
import NavSideBar from '../../layout/NavSideBar'
import WritePostBody from './WritePostBody'
import Footer from '../../layout/Footer'

const WritePost = ()=>(
    <div>
        <NavSideBar>
            <WritePostBody />
        </NavSideBar>
        <Footer />
    </div>
);

export default WritePost;