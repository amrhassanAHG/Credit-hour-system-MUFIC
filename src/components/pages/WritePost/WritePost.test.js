import React from 'react';
import renderer from 'react-test-renderer';
import WritePost from './WritePost';
import WritePostBody from './WritePostBody';

test('Writing post page Testing', () => {
    const writePostComponent = renderer.create(
      <WritePost />,
    );
    let tree = writePostComponent.toJSON();
    expect(tree).toMatchSnapshot();  
});

test('Writing post body page Testing', () => {
    const writePostBodyComponent = renderer.create(
        <WritePostBody />,
    );
    let tree = writePostBodyComponent.toJSON();
    expect(tree).toMatchSnapshot();
});