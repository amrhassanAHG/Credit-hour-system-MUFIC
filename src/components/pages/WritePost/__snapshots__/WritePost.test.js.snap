// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Writing post body page Testing 1`] = `
<div
  className="content-wrapper"
>
  <div
    className="content-header"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row mb-2"
      >
        <div
          className="col-sm-6"
        >
          <h1
            className="m-0 text-dark"
          >
            Write post
          </h1>
        </div>
        <div
          className="col-sm-6"
        >
          <ol
            className="breadcrumb float-sm-right"
          >
            <li
              className="breadcrumb-item"
            >
              Write post
            </li>
          </ol>
        </div>
      </div>
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="container-fluid"
    >
      <div
        className="card card-primary"
      >
        <div
          className="card-header"
        >
          <h3
            className="card-title"
          >
            Write Post
          </h3>
        </div>
        <form
          className="form form-group"
          onSubmit={[Function]}
          role="form"
        >
          <div
            className="card-body"
          >
            <div
              className="mb-3"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <input
                    className="form-control writePost-item"
                    onChange={[Function]}
                    placeholder="Title"
                    type="tel"
                  />
                </div>
                <div
                  className="col"
                >
                  <div
                    className="custom-file"
                  >
                    <input
                      accept=".jpg, .jpeg, .png"
                      alt=""
                      className="custom-file-input  writePost-item"
                      id="inputGroupFile01"
                      onChange={[Function]}
                      type="file"
                    />
                    <label
                      className="custom-file-label"
                      htmlFor="inputGroupFile01"
                      id="fileName"
                    >
                      Choose image
                    </label>
                  </div>
                </div>
              </div>
              <br />
              <textarea
                className="textarea"
                placeholder="Place some text here"
                style={
                  Object {
                    "border": "1px solid #dddddd",
                    "fontSize": "14px",
                    "height": "200px",
                    "lineHeight": "18px",
                    "padding": "10px",
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
          <div
            className="card-footer"
          >
            <button
              className="btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Writing post page Testing 1`] = `
<div>
  <div
    className="wrapper"
  >
    <nav
      className="main-header navbar navbar-expand navbar-white navbar-light"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            data-widget="pushmenu"
            href="#"
            role="button"
          >
            <i
              className="fas fa-bars"
            />
          </a>
        </li>
      </ul>
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item d-none d-sm-inline-block"
        >
          <a
            className="nav-link"
            href="/login"
          >
            Login
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            data-widget="pushmenu"
            href="#"
            role="button"
          >
            <i
              className="fas fa-bars"
            />
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="main-sidebar sidebar-dark-primary elevation-4"
    >
      <div>
        <a
          className="brand-link"
        >
          <img
            alt="AdminLTE Logo"
            className="img-circle elevation-3"
            height="60"
            src="logo.jpg"
            style={
              Object {
                "opacity": 0.8,
              }
            }
            width="60"
          />
          <span
            className="brand-text font-weight-dark"
            style={
              Object {
                "marginLeft": "10px",
              }
            }
          >
            MUFIC
          </span>
        </a>
        <div
          className="sidebar"
        >
          <nav
            className="mt-2"
          >
            <ul
              className="nav nav-pills nav-sidebar flex-column"
              data-accordion="false"
              data-widget="treeview"
              role="menu"
            >
              <li
                className="nav-header"
              />
            </ul>
          </nav>
        </div>
      </div>
    </div>
    <div
      className="content-wrapper"
    >
      <div
        className="content-header"
      >
        <div
          className="container-fluid"
        >
          <div
            className="row mb-2"
          >
            <div
              className="col-sm-6"
            >
              <h1
                className="m-0 text-dark"
              >
                Write post
              </h1>
            </div>
            <div
              className="col-sm-6"
            >
              <ol
                className="breadcrumb float-sm-right"
              >
                <li
                  className="breadcrumb-item"
                >
                  Write post
                </li>
              </ol>
            </div>
          </div>
        </div>
      </div>
      <div
        className="content"
      >
        <div
          className="container-fluid"
        >
          <div
            className="card card-primary"
          >
            <div
              className="card-header"
            >
              <h3
                className="card-title"
              >
                Write Post
              </h3>
            </div>
            <form
              className="form form-group"
              onSubmit={[Function]}
              role="form"
            >
              <div
                className="card-body"
              >
                <div
                  className="mb-3"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col"
                    >
                      <input
                        className="form-control writePost-item"
                        onChange={[Function]}
                        placeholder="Title"
                        type="tel"
                      />
                    </div>
                    <div
                      className="col"
                    >
                      <div
                        className="custom-file"
                      >
                        <input
                          accept=".jpg, .jpeg, .png"
                          alt=""
                          className="custom-file-input  writePost-item"
                          id="inputGroupFile01"
                          onChange={[Function]}
                          type="file"
                        />
                        <label
                          className="custom-file-label"
                          htmlFor="inputGroupFile01"
                          id="fileName"
                        >
                          Choose image
                        </label>
                      </div>
                    </div>
                  </div>
                  <br />
                  <textarea
                    className="textarea"
                    placeholder="Place some text here"
                    style={
                      Object {
                        "border": "1px solid #dddddd",
                        "fontSize": "14px",
                        "height": "200px",
                        "lineHeight": "18px",
                        "padding": "10px",
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="card-footer"
              >
                <button
                  className="btn btn-primary"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="main-footer"
  >
    <strong>
      Copyright Â© 2020 
      <a
        href="/"
      >
        FCI Menofia University
      </a>
      .
    </strong>
    <div
      className="float-right d-none d-sm-inline-block"
    >
      All rights reserved.
    </div>
  </footer>
</div>
`;

exports[`Writing post page Testing 2`] = `
<div
  className="content-wrapper"
>
  <div
    className="content-header"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row mb-2"
      >
        <div
          className="col-sm-6"
        >
          <h1
            className="m-0 text-dark"
          >
            Write post
          </h1>
        </div>
        <div
          className="col-sm-6"
        >
          <ol
            className="breadcrumb float-sm-right"
          >
            <li
              className="breadcrumb-item"
            >
              Write post
            </li>
          </ol>
        </div>
      </div>
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="container-fluid"
    >
      <div
        className="card card-primary"
      >
        <div
          className="card-header"
        >
          <h3
            className="card-title"
          >
            Write Post
          </h3>
        </div>
        <form
          className="form form-group"
          onSubmit={[Function]}
          role="form"
        >
          <div
            className="card-body"
          >
            <div
              className="mb-3"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <input
                    className="form-control writePost-item"
                    onChange={[Function]}
                    placeholder="Title"
                    type="tel"
                  />
                </div>
                <div
                  className="col"
                >
                  <div
                    className="custom-file"
                  >
                    <input
                      accept=".jpg, .jpeg, .png"
                      alt=""
                      className="custom-file-input  writePost-item"
                      id="inputGroupFile01"
                      onChange={[Function]}
                      type="file"
                    />
                    <label
                      className="custom-file-label"
                      htmlFor="inputGroupFile01"
                      id="fileName"
                    >
                      Choose image
                    </label>
                  </div>
                </div>
              </div>
              <br />
              <textarea
                className="textarea"
                placeholder="Place some text here"
                style={
                  Object {
                    "border": "1px solid #dddddd",
                    "fontSize": "14px",
                    "height": "200px",
                    "lineHeight": "18px",
                    "padding": "10px",
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
          <div
            className="card-footer"
          >
            <button
              className="btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
